File: 504-base-7/base-7.go
func convertToBase7(num int) string {
    if num == 0 {
		return "0"
	}

	isNegative := false
	if num < 0 {
		isNegative = true
		num = -num
	}

	result := ""
	for num > 0 {
		remainder := num % 7
		result = strconv.Itoa(remainder) + result
		num /= 7
	}

	if isNegative {
		return "-" + result
	}

	return result
}

File: 504-base-7/base-7.py
class Solution:
    def convertToBase7(self, num: int) -> str:
        if num == 0:
            return "0"
        
        isNegative = False
        if num < 0:
            isNegative = True

        num = abs(num)
        res = ""
        while num > 0:
            rem = num % 7
            res = str(rem) + res
            num //= 7

        if isNegative:
            return "-" + res
        else:
            return res